### PR TITLE
PBS not working outside peoples home directories (HPC only)

### PR DESCRIPTION
**Pull request by milljm**
The HPC environment, the way it currently stands, does not honor the PBS -o | -e options when the user is operating _outside_ of their home directory... So, we must allow the TestHarness to know when it is using either the emulator, or the real deal. Which is a bit unfortunate, because we wanted the TestHarness to be un-aware of any PBS emulation occurring. HPC staff claims they have a fix, and perhaps we will revert this commit when that happens.

During the testing phase, it was discovered that the PBS emulator writes to 'working directory' if _something_ is set (ex: #PBS -o foo).

Refs #6182

### CODE REVIEW

File: python/TestHarness/TestHarness.py [hunk 1]
```diff
@@ -397,6 +401,20 @@ def checkExpectError(self, output, expect_error):
...
 
 # PBS Defs
+
+  def checkPBSEmulator(self):
+    try:
+      qstat_process = subprocess.Popen(['qstat', '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+      qstat_output = qstat_process.communicate()
+    except OSError:
+      # qstat binary is not available
+      return False
```
Comment (permcody) [inline python/TestHarness/TestHarness.py]: I wonder if we shouldn't actually error out all together and tell the user that we can't find PBS. If you don't here, then they'll get an error later when the cluster launcher tries to run `qsub` or the TestHarness tries to run `qstat`. Maybe just print an error message and quit?

File: python/TestHarness/testers/RunApp.py [hunk 2]
```diff
@@ -200,6 +200,15 @@ def getPBSCommand(self, options):
     # Convert TEST_NAME to input tests file name (normally just 'tests')
     self.specs['no_copy'] = options.input_file_name
 
+    # Are we using the PBS Emulator, so make this param valid.
+    # The cluster_launcher will fill in the rest
+    if options.PBSEmulator:
+      self.specs['pbs_stdout'] = 'PBS_EMULATOR'
+      self.specs['pbs_stderr'] = 'PBS_EMULATOR'
+    else:
```
Comment (permcody) [inline python/TestHarness/testers/RunApp.py]: Just get rid of this `else` clause. We don't need to populate empty parameters. The ClusterLauncher will do the right thing when those parameters are just not supplied.

File: python/TestHarness/TestHarness.py (hunk 1 above)
Comment (milljm) [inline python/TestHarness/TestHarness.py]: Sounds good to me

File: python/TestHarness/testers/RunApp.py (hunk 2 above)
Comment (milljm) [inline python/TestHarness/testers/RunApp.py]: I'll remove those lines and commit a change

**Pull request merged**